checkov/policies_3d/checks_parser.py

```diff
@@ -18,3 +37,121 @@ def parse_raw_check(self, raw_check: Dict[str, Dict[str, Any]], **kwargs: Any) -
         check.guideline = raw_check.get("metadata", {}).get("guideline")
 
         return check
+
+    def _parse_check_v1(self, iac_records: list[Record], secrets_records: list[Record], cves_reports: list[ReportCVE]) -> Base3dPolicyCheck:
+        check = Base3dPolicyCheck()
+        self._fill_check_metadata(check)
+        check.predicaments = []
+
+        cve_predicaments = list(filter(None, [self._create_cve_predicament(cve_report) for cve_report in cves_reports]))
+        iac_predicaments = list(filter(None, [self._create_iac_predicament(iac_record) for iac_record in iac_records]))
+        secrets_predicaments = list(filter(None, [self._create_secrets_predicament(secrets_record) for secrets_record in secrets_records]))
+
+        # Generating all predicaments combinations while filtering empty lists
+        all_combinations = list(itertools.product(*filter(bool, [cve_predicaments, iac_predicaments, secrets_predicaments])))
+
+        for combination in all_combinations:
+            check.predicaments.append(
+                Predicament(
+                    logical_op='and',
+                    predicaments=[predicament for predicament in combination]
+                )
+            )
+
+        return check
+
+
+    def _create_cve_predicament(self, cve_report: ReportCVE) -> Predicament | None:
```

Comment:
It looks like the 3 functions of `_create_x_predicament` can be consolidated into one, as most of the logic is similar. So it can just be a function called `_create_predicament` and then handle the differences by a type argument or some mappings.
Another option is to create the abstract function for most of the similarity, and let each of these specific functions call the abstract one with specific values. makes sense?

Reply:
extracted to `_create_module_predicament`